inference-engine/src/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp

```diff
@@ -16,53 +16,61 @@
 NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertInterpolate1ToInterpolate4, "ConvertInterpolate1ToInterpolate4", 0);
 
 ngraph::pass::ConvertInterpolate1ToInterpolate4::ConvertInterpolate1ToInterpolate4() {
-    MATCHER_SCOPE(ConvertInterpolate1ToInterpolate4);
-    auto interpolate1 = ngraph::pattern::wrap_type<ngraph::opset1::Interpolate>({pattern::any_input(pattern::has_static_rank()), pattern::any_input()});
+    auto interpolate1 = ngraph::pattern::wrap_type<ngraph::opset1::Interpolate>({pattern::any_input(pattern::has_static_shape()), pattern::any_input()});
     ngraph::matcher_pass_callback callback = [this](pattern::Matcher& m) {
-        auto interpolate1 = std::dynamic_pointer_cast<ngraph::opset1::Interpolate>(m.get_match_root());
-        if (!interpolate1)
+        auto interpolationV0 = std::dynamic_pointer_cast<ngraph::opset1::Interpolate>(m.get_match_root());
+        if (!interpolationV0) {
             return false;
+        }
 
-        auto interpolate_attrs = interpolate1->get_attrs();
-        auto input_shape_rank = interpolate1->input(0).get_partial_shape().rank().get_length();
+        auto& inp_partial_shape = interpolationV0->get_input_partial_shape(0);
+        auto& out_shape = interpolationV0->get_output_shape(0);
+        auto attrsV0 = interpolationV0->get_attrs();
 
-        // attrs
-        auto mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode();
-        if (interpolate_attrs.mode == "nearest") {
-            mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::nearest;
-        } else if (interpolate_attrs.mode == "cubic") {
-            mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::cubic;
-        } else if (interpolate_attrs.mode == "linear") {
-            if (input_shape_rank < 5) {
-                mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::linear_onnx;
-            } else if (input_shape_rank == 5) {
-                mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::linear;
-            } else {
-                return false;
+        std::vector<float> scales(attrsV0.axes.size(), 1.0f);
+        if (inp_partial_shape.is_static()) {
```

Comment:
Now you always check in pattern that input shape is static. So you either don't need this check in pattern or you need to remove this check.

Reply:
Fixed.